sycl/source/detail/graph_impl.cpp

```diff
@@ -418,8 +419,9 @@ void command_graph<graph_state::modifiable>::make_edge(node Sender,
 template <>
 command_graph<graph_state::executable>
 command_graph<graph_state::modifiable>::finalize(
-    const sycl::context &CTX, const sycl::property_list &) const {
-  return command_graph<graph_state::executable>{this->impl, CTX};
+    const sycl::property_list &) const {
+  return command_graph<graph_state::executable>{this->impl,
+                                                this->impl->get_context()};
```

Comment:
I've removed the context member from `command_graph<executable>` although not mentioned here as it should be using the one from the `exec_graph_impl`.

For the `exec_graph_impl` member I would personally lean towards keeping it, though it could be inferred as you say. I think the context is directly relevant to the class so keeping it around as a member variable for convenience makes sense to me and is a bit nicer when you do need to use it (which may be more often as the implementation grows more complex). Plus its not expensive to keep around and there's no concerns about shared ownership here. I'm happy to remove it if you feel strongly about it though.

Reply:
I don't feel strongly about it, was just something that caught my eye, so happy to keep it around.